src/OpenAPI/Filter/PathMatcher.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Membrane\OpenAPI\Filter;
+
+use Membrane\Filter;
+use Membrane\OpenAPI\PathMatcher as PathMatcherClass;
+use Membrane\Result\Message;
+use Membrane\Result\MessageSet;
+use Membrane\Result\Result;
+
+class PathMatcher implements Filter
+{
+    public function __construct(private readonly PathMatcherClass $pathMatcher)
+    {
+    }
+
+    public function filter(mixed $value): Result
+    {
+        if (!is_string($value)) {
+            return Result::invalid(
+                $value,
+                new MessageSet(null, new Message('PathMatcher filter expects string, %s passed', [gettype($value)]))
+            );
+        }
+
+        return Result::noResult($this->pathMatcher->getPathParams($value));
```

Comment:
This method can throw an exception if the path fails to match; will need to catch it and return an invalid result instead.